sympy/solvers/solvers.py

```diff
@@ -694,6 +734,10 @@ def _sympified_list(w):
         elif isinstance(fi, (bool, C.BooleanAtom)) or fi.is_Relational:
             return reduce_inequalities(f, symbols=symbols)
```

Comment:
Yes, checking should be done. Let's leave that to #8715, however. I just wanted to reorganize and enhance unrad in this PR. I've retracted most of the significant unrad changes to make review of this somewhat easier. Thanks for looking this over.